datafusion/core/src/datasource/physical_plan/parquet/row_groups.rs

```diff
@@ -168,93 +174,147 @@ pub(crate) async fn prune_row_groups_by_bloom_filters<
 }
 
 struct BloomFilterPruningPredicate {
-    /// Actual pruning predicate (rewritten in terms of column min/max statistics)
-    predicate_expr: Option<phys_expr::BinaryExpr>,
+    /// The input schema against which the predicate will be evaluated
+    schema: SchemaRef,
+
+    ///  predicate
+    predicate_expr: Arc<dyn PhysicalExpr>,
     /// The statistics required to evaluate this predicate
     required_columns: Vec<String>,
 }
 
 impl BloomFilterPruningPredicate {
-    fn try_new(expr: &Arc<dyn PhysicalExpr>) -> Result<Self> {
-        let binary_expr = expr.as_any().downcast_ref::<phys_expr::BinaryExpr>();
-        match binary_expr {
-            Some(binary_expr) => {
-                let columns = Self::get_predicate_columns(expr);
-                Ok(Self {
-                    predicate_expr: Some(binary_expr.clone()),
-                    required_columns: columns.into_iter().collect(),
-                })
-            }
-            None => Err(DataFusionError::Execution(
+    fn try_new(schema: SchemaRef, expr: &Arc<dyn PhysicalExpr>) -> Result<Self> {
+        let columns = Self::get_predicate_columns(expr);
+        if columns.is_empty() {
+            return Err(DataFusionError::Execution(
                 "BloomFilterPruningPredicate only support binary expr".to_string(),
-            )),
+            ));
         }
-    }
 
-    fn prune(&self, column_sbbf: &HashMap<String, Sbbf>) -> bool {
-        Self::prune_expr_with_bloom_filter(self.predicate_expr.as_ref(), column_sbbf)
+        Ok(Self {
+            schema,
+            predicate_expr: expr.clone(),
+            required_columns: columns.into_iter().collect(),
+        })
     }
 
     /// filter the expr with bloom filter return true if the expr can be pruned
-    fn prune_expr_with_bloom_filter(
-        expr: Option<&phys_expr::BinaryExpr>,
-        column_sbbf: &HashMap<String, Sbbf>,
-    ) -> bool {
-        if expr.is_none() {
+    fn prune(&self, column_sbbf: &HashMap<String, Sbbf>) -> bool {
+        // rewrite any <col = literal> exprs to to `false` if we can provve they are using the bloom filter.
```

Comment:
I am quite pleased with this forumulation -- namely rewrite the physical expression using information from the bloom filter and then apply predicate propagation to see if we can determine if the predicate is always true